books/tests.py

```diff
@@ -137,3 +140,77 @@ def test_can_create_book_with_cc_license(self):
                     )
         book_index.add_child(instance=book)
         self.assertEqual(book.license_url, 'https://creativecommons.org/licenses/by/4.0/')
+
+
+class ResourceTests(WagtailPageTests):
+    def setUp(self):
+        pass
+
+    @classmethod
+    def setUpTestData(cls):
+        # create root page
+        root_page = Page.objects.get(title="Root")
+        # create homepage
+        homepage = HomePage(title="Hello World",
+                            slug="hello-world",
+                            )
+        # add homepage to root page
+        root_page.add_child(instance=homepage)
+        # create book index page
+        book_index = BookIndex(title="Book Index",
+                               page_description="Test",
+                               dev_standard_1_description="Test",
+                               dev_standard_2_description="Test",
+                               dev_standard_3_description="Test",
+                               dev_standard_4_description="Test",
+                               )
+        # add book index to homepage
+        homepage.add_child(instance=book_index)
+
+        test_image = SimpleUploadedFile(name='openstax.png',
+                                        content=open("oxauth/static/images/openstax.png", 'rb').read())
+        cls.test_doc = Document.objects.create(title='Test Doc', file=test_image)
+
+        cls.book_index = Page.objects.get(id=book_index.id)
+
+    def test_faculty_resources_available(self):
+        book_index = BookIndex.objects.all()[0]
+        root_page = Page.objects.get(title="Root")
+        faculty_resource = snippets.models.FacultyResource(heading="Instructor Getting Started Guide",
+                                                           description="<p data-block-key=\"6o2yl\">Download our helpful guide to all things OpenStax.<br/></p>",
+                                                           unlocked_resource=True,
+                                                           creator_fest_resource=False)
+        faculty_resource.save()
+        resource = books.models.FacultyResources(link_external="https://openstax.org",
+                                                 link_text="Go!",
+                                                 coming_soon_text=None,
+                                                 video_reference_number=None,
+                                                 updated=None,
+                                                 featured=False,
+                                                 k12=False,
+                                                 print_link=None,
+                                                 resource_id=faculty_resource.id)
+        resource.save()
+
+        # book_faculty_resource = books.models.BookFacultyResources(faculty_resources_ptr_id=resource.id,
```

Comment:
you probably want to use `faculty_resource_ptr`... the id might not be necessary?

Reply:
then just pass the object to it `resource`, instead of `resource.id`